### PR TITLE
tensorboard

### PR DESCRIPTION
**Pull request by sauravkdeo**

### CODE REVIEW

**Comment by FateScript** [comment]
Could you plz tell me what is the purpose of this PR?

**Comment by sauravkdeo** [comment]
@FateScript 
This will add the training loss graph and the learning rate graph to the tensorboard.

**Comment by mahnehsilla** [comment]
Interesting, I do not know whether the code is correct, but would certainly be useful to have the loss and maybe also metrics like MSE as well, wouldnt it ?

File: yolox/core/trainer.py [hunk 1]
```diff
@@ -265,6 +265,10 @@ def after_iter(self):
             )
 
             if self.rank == 0:
+                if self.args.logger == "tensorboard":
+                    self.tblogger.add_scalar("train/lr", self.meter["lr"].latest, self.epoch + 1)
+                    for k, v in loss_meter.items():
+                        self.tblogger.add_scalar(f"train/{k}", v.latest, self.epoch + 1)
```
Comment (FateScript) [inline yolox/core/trainer.py:271]: IMHO, It should be self.iter here, not epoch. WDYT @sauravkdeo

**Review by FateScript — CHANGES_REQUESTED** [review]
PTAL @sauravkdeo

File: yolox/core/trainer.py (hunk 1 above)
Comment (sauravkdeo) [reply yolox/core/trainer.py]: yes, You are right, `self.progress_in_iter` will capture more granular details.

**Comment by sauravkumar-volt** [comment]
@FateScript any updates on this

**Comment by FateScript** [comment]
LGTM.

**Review by FateScript — APPROVED** [review]

**Comment by FateScript** [comment]
@sauravkdeo  workflow failed here, plz lint your code. Thanks.

**Comment by sauravkdeo** [comment]
@FateScript  Done

**Comment by FateScript** [comment]
@sauravkdeo  Too much code is modified in this PR, plz carefully lint your own modified code but not other part in the file. Thanks.

**Comment by FateScript** [comment]
Code still failed with lint @sauravkdeo

**Comment by sauravkdeo** [comment]
@FateScript Done

**Comment by sauravkdeo** [comment]
@FateScript  Can you please approve this?
